.github/workflows/unit_tests.yml

```diff
@@ -1,217 +1,212 @@
-name: "unit_tests"
+name: unit tests
 concurrency:
   group: ${{ github.workflow }}-${{ github.event.number }}-${{ github.event.type }}
   cancel-in-progress: true
 on:
-  push:
-    branches:
-      - "main"
   pull_request:
-    branches:
-      - "*"
+    paths:
+      - '**.py'
+  push:
+    branches: [main]
+    paths:
+      - '**.py'
+  workflow_dispatch:
 
 jobs:
-  # Run style tests
   style:
-    runs-on: ${{ matrix.os }}
-    strategy:
-      fail-fast: false
-      matrix:
-        os: [ubuntu-latest]
-        python-version: ["3.10"]
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - name: Set up Python ${{ matrix.python-version }}
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Setup Python 3.9
         uses: actions/setup-python@v4
         with:
-          python-version: ${{ matrix.python-version }}
-      # install only the dependencies required for style
-      - name: Install pip
+          python-version: '3.9'
+          architecture: 'x64'
+      - name: Install dependencies
         run: |
-          pip install --upgrade --progress-bar off pip setuptools wheel
-          pip install --upgrade --upgrade-strategy eager --progress-bar off black check-manifest codespell flake8 flake8-array-spacing isort mypy pydocstyle
-      - name: Run style & documentation tests
-        run: make run-checks
+          python -m pip install --progress-bar off --upgrade pip setuptools wheel
+          python -m pip install --progress-bar off .[style]
+      - name: Setup flake8 annotations
+        uses: rbialon/flake8-annotations@v1
+      - name: Run flake8
+        uses: py-actions/flake8@v2
+        with:
+          path: "mne_icalabel"
+      - name: Run isort
+        uses: jamescurtin/isort-action@master
```

Comment:
For these, I'm not 100% knowledgeable, is this the official version? E.g. I found this on the web: https://github.com/isort/isort-action.

Same for `flake8` and `mypy`?

Reply:
I'm guessing `jamescurtin/isort-action` got renamed/moved to the `isort` namespace.. as I can't find anymore the GitHub page of `jamescurtin/isort-action` and as he his the main (almost only) contributor to that action ;)

Also: https://pycqa.github.io/isort/docs/configuration/github_action.html
Which does point to the new link, but I'm 99% sure it was pointing to jamescurtin namespace before ;)

I'll check them and update them.